ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -0,0 +1,707 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.tests
+
+import ai.x.diff.conversions._
+import com.daml.ledger.api.testtool.infrastructure.{LedgerSession, LedgerTest, LedgerTestSuite}
+import com.digitalasset.ledger.api.v1.value.{RecordField, Value}
+import com.digitalasset.ledger.client.binding.Value.encode
+import com.digitalasset.ledger.client.binding.Primitive
+import com.digitalasset.ledger.test_stable.Test.Agreement._
+import com.digitalasset.ledger.test_stable.Test.AgreementFactory._
+import com.digitalasset.ledger.test_stable.Test.Choice1._
+import com.digitalasset.ledger.test_stable.Test.Dummy._
+import com.digitalasset.ledger.test_stable.Test.DummyFactory._
+import com.digitalasset.ledger.test_stable.Test.ParameterShowcase._
+import com.digitalasset.ledger.test_stable.Test.TriProposal._
+import com.digitalasset.ledger.test_stable.Test._
+import io.grpc.Status
+import scalaz.Tag
+
+import scala.concurrent.Future
+
+class TransactionService(session: LedgerSession) extends LedgerTestSuite(session) {
+
+  private[this] val beginToBeginShouldBeEmpty =
+    LedgerTest(
+      "TXBeginToBegin",
+      "An empty stream should be served when getting transactions from and to the beginning of the ledger") {
+      ledger =>
+        for {
+          party <- ledger.allocateParty()
+          request = ledger.getTransactionsRequest(Seq(party))
+          fromAndToBegin = request.update(_.begin := ledger.begin, _.end := ledger.begin)
+          transactions <- ledger.flatTransactions(fromAndToBegin)
+        } yield {
+          assert(
+            transactions.isEmpty,
+            s"Received a non-empty stream with ${transactions.size} transactions in it.")
+        }
+    }
+
+  private[this] val serveElementsUntilCancellation =
+    LedgerTest("TXServeUntilCancellation", "Items should be served until the client cancels") {
+      ledger =>
+        val transactionsToSubmit = 14
+        val transactionsToRead = 10
+        for {
+          party <- ledger.allocateParty()
+          dummies <- Future.sequence(Vector.fill(transactionsToSubmit)(for {
```

Comment:
you don't need a for-comprehension with one "producer" that directly yields the result.
This is equivalent:
```scala
Future.sequence(Vector.fill(transactionsToSubmit)(ledger.create(party, Dummy(party))))
```

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2713/commits/a46d2a3fba1d5f92154c4116751abbd30056ba59